packages/core/chrome/core-chrome-browser-internal/src/ui/header/header.tsx

```diff
@@ -97,6 +97,7 @@ export function Header({
       <>
         <LoadingIndicator loadingCount$={observables.loadingCount$} showAsBar />
         <HeaderTopBanner headerBanner$={observables.headerBanner$} />
+        <EuiHeader theme="dark" position="fixed" style={{ display: 'none' }} />
```

Comment:
I won't lie, this is a clever "hack" 😅 As a heads up, this may be less performant than rolling your own `useEffect`... but since we may potentially be removing the `.euiBody--headerIsFixed` static className when we convert `EuiHeader` to CSS-in-JS, this might actually be the most future-proof approach after all. No objections from the EUI team on this one, not sure about the Kibana folks.

Reply:
For the record:

Full-height content with banner are supposed to be leveraging the `kibanaFullBodyHeight` mixin:

https://github.com/elastic/kibana/blob/570dcc07b52245e44cd4b43679e839673b7c7eaa/src/core/public/_mixins.scss#L3-L4

that is using the classes automatically added to the document's body:

https://github.com/elastic/kibana/blob/4522e04287c2b8b92ec49acd7ec6f31d59622733/packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx#L134-L143

And it was working at some point.

So when I read the issue's description:

> The bug happens in embed mode because padding-top offset style is coming from the EuiHeader styles (.euiBody--headerIsFixed class on body)

I assume something did broke in the EUI styles at some point. 

@Dosant do you think it would be possible to somehow fix the existing mixins instead of introducing this work around?

Reply:
> I assume something did broke in the EUI styles at some point. The 'layout' styles shouldn't be coming from EUI directly here...

EUI has not converted any `.euiBody` classes that relate to headers to Emotion, they are all still globally/statically available via Sass/CSS. I would be more inclined to look at the Kibana mixins to debug.

Reply:
Sorry for not replying. Was away. 

@pgayvallet, 

> Full-height content with banner are supposed to be leveraging the kibanaFullBodyHeight mixin:

Are you saying that apps which have content for embedding should make sure their page layouts are using `kibanaFullBodyHeight` mixing ?  If so, then looks like dashboard app isn't using this mixing. So does this mean we should apply the fix on dashboard side?


Otherwise currently it looks like a style dependency from kibana's global css to eui's header presence css is coming from here: 


https://github.com/elastic/kibana/blob/1e90bcdb2204b169512127e0afa6068c9cb790d0/src/core/public/styles/rendering/_base.scss#L38-L39

-> 

https://github.com/elastic/eui/blob/c9c9623a9dad150d0683bbc38fbd3d3312d929d0/src/global_styling/mixins/_header.scss#L6-L7